lms/views/basic_lti_launch.py

```diff
@@ -24,6 +24,7 @@
     add_user_to_group,
     report_lti_launch,
     upsert_course_group,
+    upsert_group_info,
```

Comment:
It seems like we want this everywhere we use `upsert_course_group`. Is the freedom we give ourselves by having them independent just the freedom to make a mistake by including one and not other other?

Reply:
I think the place to move this might be into a new `lms/views/decorators/_helpers.py` module (a module containing private helpers for the view decorators). The reason is that `upsert_group_info()` would then be part of the public API of this `_helpers.py`, which means that there is a place to put unit tests that call `upsert_group_info()` directly (in `tests/lms/views/decorators/_helpers_test.py`) like its current unit tests do. Rather than having to test it indirectly via the `upsert_course_group()` decorator that uses it (then `upsert_course_group()`'s tests get too long). This also means that `upsert_group_info()` is potentially reusable, though I don't know that we ever will reuse it.

```python
from lms.views.decorators._helpers import upsert_group_info

def upsert_course_group(wrapped):
    @functools.wraps(wrapped)
    def wrapper(context, request):
        ...
        upsert_group_info(...)
        ...
```

This might seem like too deep a hierarchy (not only do views have decorators but the view decorators now have helpers). On the other hand it kind of makes sense: the view decorators themselves are Pyramid integration stuff, so it makes sense that they should do little else than call helper functions (or services or whatever, some non-Pyramid-connected code).